### DISCUSSION TITLE
About sse-s3 encryption

### DISCUSSION THREAD

**Discussion by alixiaodi** [minio/minio#17532] [Q&A] [answered]
Is there a more detailed document on the encryption method of sse-s3? I don't know how to operate it, and it seems to be too complicated to operate at present

**Comment by ravindk89** [comment]
Have you already looked at our [documentation](https://min.io/docs/minio/linux/administration/server-side-encryption/server-side-encryption-sse-s3.html) ? 

What is the complexity you are encountering? If it's around [setting up the external KMS](https://min.io/docs/minio/linux/operations/server-side-encryption/configure-minio-kes-hashicorp.html) that is hard to avoid - our lower-bound of complexity is based on what is required to set up the eKMS.

**Reply by alixiaodi** [reply]
I have read your official website document, if I want to use sse-s3 encryption, but I do not want to use cloud service kms, I want to use my own deployment or other open source products, how to use, or how to deploy kms without a clear document, I need to search by myself, I can not quickly use minio to integrate

**Reply by alixiaodi** [reply]
I want a foolproof deployment document that I can set up by following the steps, which doesn't look very good right now

**Reply by alixiaodi** [reply]
Do you have tutorial videos on youtube

**Reply by klauspost** [reply, marked as answer]
KES can be used [standalone](https://github.com/minio/kes/wiki/Filesystem-Keystore), but for (hopefully) obvious reasons there are no remote secrets.

**Reply by ravindk89** [reply]
@alixiaodi I can appreciate finding written instructions confusing, but you aren't telling us what part of that tutorial you have found difficult, where your stumbling block is, or at which point you can no longer proceed.

If what you want is a way to configure SSE-S3 without any external key management, which we cannot recommend for any serious deployment, you can use the `MINIO_KMS_SECRET_KEY=my-minio-key:<base64-encoded-256-bit-string>`

Then you can enable SSE-S3 using the documented commands. But anyone with access to your host machine can pull that envvar and access that key for the purpose of decryption. It is 100% only for testing and learning.

**Reply by alixiaodi** [reply]
Let me try your way first, my problem now is to build a kms server, I can't understand haha, I tried to set up sse-s3 encryption in the bucket, he now reminds me that I did not point to the kms server, I do not understand the kms server at all
